# Welcome to scVMAP Platform Development Front-end!

> scVMAP: https://bio.liclab.net/scvmap/

> scVMAP reproducibility:: https://github.com/YuZhengM/scvmap_reproducibility

> scVMAP tutorial: https://scvmap.readthedocs.io/en/latest/

> scVMAP back-end: https://github.com/YuZhengM/scvmap

> scVMAP API: https://bio.liclab.net/scvmap_service/scvmap.html

A comprehensive platform for **integrating** `single-cell chromatin accessibility regions` with `causal variants`.

![overview.png](src/assets/img/home/overview.png)

## Introduction

scVMAP is a user-centric database that offers intuitive workflows, customizable parameters, and comprehensive data accessibility, featuring:

1. Integrates causal variants with chromatin accessibility regions at single-cell resolution.
2. Includes data from 183 scATAC-seq samples and 15,884 fine-mapping datasets.
3. Encompasses over 3,876,800 effective trait-sample pairs, 26.3 million trait-cell type pairs, and 32.1 billion trait-cell pairs.
4. Performs enrichment analysis for traits across specific cells or cell types.
5. Identifies relevant genes and transcription factors (TFs), and performs pathway enrichment analysis for these genes in the context of specific samples and traits.
6. Constructs an integrated regulatory network (variant-TF-gene) linking traits to samples.
7. Offers comprehensive resources for data download.
8. Makes the code for reproducing the database workflow publicly available.
9. Publishes the full-stack development code for the web application.
10. Provides API interfaces for programmatic data access.

## Implementation

> Node.js: v16.13.0

> Vue: 3.2.4

> Axios: 0.21.4

> Bootstrap: v5.1.3

> Element-UI: (element-plus 2.2.0)

> Font Awesome: 6.1.1

> Echarts: 5.3.1

> Plotly: 2.23.0

> CanvasXpress: 38.4.1

## Deploy

[README.md](deploy/README.md)
